### PR TITLE
Multivariable Polynomials now mutable

### PR DESCRIPTION
**Pull request by myluszczak**
@isuruf @shivamvats @StoicBronco @irislq @chenchfort This PR contains updated multivariable polynomials.  The MultivariatePolynomialExpr and MultivariateIntPolynomialExpr are now mutable, and MultivariatePolynomialExpr now has overloaded operators.

### CODE REVIEW

File: symengine/dict.cpp [hunk 1]
```diff
@@ -42,7 +42,7 @@ inline std::ostream &print_vec(std::ostream &out, T &d)
             out << ", ";
         out << *p;
     }
-    out << "}";
+    out << "]}";
```
Comment (nishnik) [inline symengine/dict.cpp]: Why this change ?

**Comment by myluszczak** [comment]
@nishnik That must have been introduced during rebasing.  It's been fixed now.

**Comment by isuruf** [comment]
Can you fix these commits. This PR seems to contain commits that have already been merged

**Comment by chenchfort** [comment]
@isuruf Rebase on top of master?

**Comment by myluszczak** [comment]
@isurf The contents of this commit are included in PR #968.  That one is on a branch off of master with a neater commit history.  I'd prefer you review and merge that one, so I'm closing this PR.
